documentation/asciidoc/topics/proc_creating_caches_hotrod.adoc

```diff
@@ -1,90 +1,71 @@
+             .templateName(DefaultTemplate.DIST_SYNC);
       builder.remoteCache("another-cache")
-               .configuration("<infinispan><cache-container><distributed-cache name=\"another-cache\"><encoding media-type=\"application/x-protostream\"/></distributed-cache></cache-container></infinispan>");
+             .configuration("<infinispan><cache-container><distributed-cache name=\"another-cache\"><encoding media-type=\"application/x-protostream\"/></distributed-cache></cache-container></infinispan>");
       try (RemoteCacheManager cacheManager = new RemoteCacheManager(builder.build())) {
-         // Get a remote cache that does not exist.
-         // Rather than return null, create the cache from a template.
-         RemoteCache<String, String> cache = cacheManager.getCache("my-cache");
-         /// Store a value.
-         cache.put("hello", "world");
-         // Retrieve the value and print it.
-         System.out.printf("key = %s\n", cache.get("hello"));
+      // Get a remote cache that does not exist.
+      // Rather than return null, create the cache from a template.
+      RemoteCache<String, String> cache = cacheManager.getCache("my-cache");
+      // Store a value.
+      cache.put("hello", "world");
+      // Retrieve the value and print it.
+      System.out.printf("key = %s\n", cache.get("hello"));
 ----
 
+This example shows how to create a cache named CacheWithXMLConfiguration using the `XMLStringConfiguration()` method to pass the cache configuration as XML:
 
+[source,java,options="nowrap",subs=attributes+]
+----
+import org.infinispan.client.hotrod.RemoteCacheManager;
+import org.infinispan.commons.configuration.XMLStringConfiguration;
+...
+
+private void createCacheWithXMLConfiguration() {
+    String cacheName = "CacheWithXMLConfiguration";
+    String xml = String.format("<infinispan>" +
+                                  "<cache-container>" +
+                                  "<distributed-cache name=\"%s\" mode=\"SYNC\">" +
+                                    "<encoding media-type=\"application/x-protostream\"/>" +
+                                    "<locking isolation=\"READ_COMMITTED\"/>" +
+                                    "<transaction mode=\"NON_XA\"/>" +
+                                    "<expiration lifespan=\"60000\" interval=\"20000\"/>" +
+                                  "</distributed-cache>" +
+                                  "</cache-container>" +
+                                "</infinispan>"
+                                , cacheName);
+    manager.administration().getOrCreateCache(cacheName, new XMLStringConfiguration(xml));
+    System.out.println("Cache with configuration exists or is created.");
+}
+----
+
+[discrete]
+== Using Hot Rod client properties
 
-.Reference
+When you invoke `cache.get()` calls for named caches that do not exist, {brandname} creates them from the Hot Rod client properties instead of returning null.
```

Comment:
this is not clear to me what is happening here.
is this a `cacheManager.getCache("my-cache")` actually and this call creates a cache from the properties?
Or is really a `cache.get("key")` ? in this case, if the cache is created with that call I'm expecting to have a `null` retuned.

Reply:
Hmm... Good catch. It is `cacheManager.getCache("my-cache");`

For example: https://github.com/infinispan/infinispan-simple-tutorials/blob/00d80cc878ebf98684e0eff2f7249adf5c6fb200/remote-per-cache/src/main/java/org/infinispan/tutorial/simple/remote/percache/InfinispanRemotePerCache.java#L46